src/coreComponents/mesh/generators/WellGeneratorBase.hpp

```diff
@@ -75,7 +75,8 @@ class WellGeneratorBase : public dataRepository::Group
   /**
    * @brief Main function of the class that generates the well geometry
    */
-  virtual void generateWellGeometry( ) = 0;
+  void generateWellGeometry( );
```

Comment:
I removed this abstraction because it is not really an abstraction. You have intermediate data structure to be filled up with data: 1. generated (InternalWell) or 2. loaded (VTKWell) and you certainly do not want to duplicate code that fills this intermediate data structure in each of these subclasses. If it is the way to go. I can add an ABC class with all these virtual stuff in it.

Reply:
Please remember that this part of the code is only a few days old and that it's surely not the way we want it to be eventually. Some part of the code will still moved and some code will disappear and may change.
Also, having a somehow homogeneous approach within a module is helpful: if we want to remove the `ABC`s I thing it can be useful to do it all at once...

Now, if there's some redundancy with `LineBlock*`, maybe we could consider putting some output into the `LineBlock` only and not the `WellGenertorBase`? I think that the `MeshGeneratorBase` is far lighter? Also, some functions are surely meant to disappear.

Reply:
Also, let's be clear: you have some very good arguments and what you're saying makes sense for sure.
It's just that we're slowly, step by step, trying to reorganize the module in a way (which might not be the correct one eventually). Having everything abiding by the same pattern will surely help if we decide to change the way we're doing things.